sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -439,6 +440,16 @@ class Analyzer(
             case s: Star => s.expand(child, resolver)
             case o => o :: Nil
           })
+        case p: Murmur3Hash if containsStar(p.children) =>
```

Comment:
@cloud-fan Now, only four cases. I did not add a new function `replaceAllChildren` for less code changes. Please let me know if you think we still need to do it. Thanks!